templates/init.go

```diff
@@ -0,0 +1,68 @@
+package templates
+
+import (
+	"embed"
+	"fmt"
+	"io/fs"
+	"path"
+
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	//go:embed charger/*.yaml meter/*.yaml vehicle/*.yaml
+	yamlTemplates embed.FS
+
+	templates = make(map[string][]Template)
+)
+
+const (
+	Charger = "charger"
+	Meter   = "meter"
+	Vehicle = "vehicle"
+)
+
+//go:generate go run generate/generate.go
+func init() {
```

Comment:
das macht eine Menge Yaml parsen beim Programmstart. Können wir das bitte nicht tun oder zur Not wenn gar nix besseres einfällt durch ein

    sync.Once.Do()

kapseln?

Reply:
Das war dein Code für das Generieren der Docs, hab da noch nichts mit gemacht. Der ist ja nur für go:generate notwendig, sonst nicht. Wie willst das haben?

Reply:
dann gerne ins generate. aktuell wird er immer aufgerufen.

Reply:
mom, da wird die Liste der Templates geholt, damit die dann auch instantiiert werden können. Ne File-liste sollte genügen, und dann das instantiieren wird ja eh geparst

Reply:
Hab mal `sync.Once` hinzugefügt. Parsing weglassen geht nicht, weil sonst die Templates nicht über den Namen sondern nur über den Dateinamen des Templates identifiziert werden könnten. Kann man machen, dann stehen die halt in den Konfigurationen drin.  Was ist dir lieber?

Reply:
dann lassen wir's mal so